### ISSUE TITLE
Improve UX of mnemonic import

### ISSUE THREAD

**Issue by pythcoiner** [wizardsardine/liana#513]
I think it can be few little improvement when importing a mnemonic:
- add shortcut (tab?) to go to the next input field
- when input a 'word' add autocomplete inline when there is only one available word (enter key for valid autocomplete + move caret to the next input field)
- when input a word not existing (or typo error) display a red cross
- on last word (the one including the "checksum")  if the word is wrong (because wrong checksum) display a red cross instead a green V
- disable the next button if the checksum is wrong

**Comment by darosior** [comment]
> add shortcut (tab?) to go to the next input field

Yeah, i think it could even have its own issue as it happens everywhere there is multiple inputs. Having to grab your mouse to click on the next input puts a lot of friction.

**Comment by darosior** [comment]
If anyone wants to tackle this, feel free to address any one of the point separately.

**Comment by jp1ac4** [comment]
> add shortcut (tab?) to go to the next input field

Hi there, I don't think this is currently possible if I understand this issue correctly: https://github.com/iced-rs/iced/issues/489.

> disable the next button if the checksum is wrong

Related to this, I noticed that if the checksum is wrong and an error appears when clicking Next, the Next button remains blocked even after the words are corrected. The change below seems to fix this, but it might not be required if other changes to the logic are made on this page so I'll wait and see what's decided first.
```diff
diff --git a/gui/src/installer/step/mnemonic.rs b/gui/src/installer/step/mnemonic.rs
index 7198b83..0bebd5d 100644
--- a/gui/src/installer/step/mnemonic.rs
+++ b/gui/src/installer/step/mnemonic.rs
@@ -94,6 +94,7 @@ impl Step for RecoverMnemonic {
                         let suggestions = self.language.words_by_prefix(&value);
                         if suggestions.contains(&value.as_ref()) {
                             *valid = true;
+                            self.error = None;
                             self.suggestions = Vec::new();
                         } else {
                             self.suggestions = suggestions.iter().map(|s| s.to_string()).collect();
```

**Comment by jp1ac4** [comment]
I'm not sure if it's currently possible as I thought that comment was regarding a way to add this functionality to iced.

**Comment by pythcoiner** [comment]
the comment is not really clear...
is there an event handler in iced, from which we can get the keyboard event?
how can we get on which widget the focus is on?

**Comment by pythcoiner** [comment]
i'm digging iced doc, i think we can use [iced::widget::text_input::move_cursor_to()](https://docs.rs/iced/0.12.1/iced/widget/text_input/fn.move_cursor_to.html#) in order to move focus between TextInput

**Comment by pythcoiner** [comment]
https://github.com/iced-rs/iced/blob/8ce16aba6204cb5c02a709cdf79c309f7b7e0196/examples/todos/src/main.rs#L145-L151

**Comment by pythcoiner** [comment]
> > add shortcut (tab?) to go to the next input field
> 
> Yeah, i think it could even have its own issue as it happens everywhere there is multiple inputs. Having to grab your mouse to click on the next input puts a lot of friction.

this is addressed by #1019
